frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx

```diff
@@ -0,0 +1,291 @@
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.mirroringhealth.classes}
+        columnID={blockPoolColumnInfo.mirroringhealth.id}
+      >
+        <Tooltip
+          content={`${t('ceph-storage-plugin~Last Synced')} ${dateTimeFormatter(lastChecked)}`}
+        >
+          <Status status={_.capitalize(mirroringHealth)} />
+        </Tooltip>
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.compressionstatus.classes}
+        columnID={blockPoolColumnInfo.compressionstatus.id}
+      >
+        {compressionStatus ? t('ceph-storage-plugin~Enabled') : t('ceph-storage-plugin~Disabled')}
+      </TableData>
+      <TableData
+        className={blockPoolColumnInfo.compressionsavings.classes}
+        columnID={blockPoolColumnInfo.compressionsavings.id}
+      >
+        {'-'}
+      </TableData>
+      <TableData className={Kebab.columnClass}>
+        <ResourceKebab
+          actions={menuActions}
+          kind={referenceFor(obj)}
+          resource={obj}
+          isDisabled={
+            obj?.metadata?.deletionTimestamp ||
+            props.cephCluster?.metadata?.name === CEPH_EXTERNAL_CR_NAME ||
+            isDefaultPool(obj)
+          }
+        />
+      </TableData>
+    </TableRow>
+  );
+};
+
+const BlockPoolList: React.FC<BlockPoolListProps> = (props) => {
+  const { t } = useTranslation();
+
+  // Hooks
+  const [storageClasses, setStorageClasses] = React.useState<OcsStorageClassKind[]>([]);
+  const [cephCluster, setCephCluster] = React.useState<CephClusterKind>();
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+  const filteredCephClusters: CephClusterKind[] = useDeepCompareMemoize(cephClusters, true);
+  const [scResources, scLoaded, scLoadError] = useK8sWatchResource<OcsStorageClassKind[]>(
+    scResource,
+  );
+  const filteredScResources: OcsStorageClassKind[] = useDeepCompareMemoize(scResources, true);
+
+  const poolNames: string[] = React.useMemo(() => props.data.map((pool) => pool.metadata?.name), [
+    props.data,
+  ]);
+
+  React.useEffect(() => {
+    if (filteredCephClusters.length && isLoaded && !loadError) {
+      setCephCluster(filteredCephClusters[0]);
+    }
+  }, [filteredCephClusters, isLoaded, loadError]);
+
+  React.useEffect(() => {
+    if (filteredScResources && scLoaded && !scLoadError) {
+      setStorageClasses(filteredScResources);
+    }
+  }, [filteredScResources, scLoaded, scLoadError]);
+
+  // Metrics
+  const [poolRawCapacityMetrics, rawCapLoadError, rawCapLoading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    query: getPoolQuery(poolNames, StorageDashboardQuery.POOL_RAW_CAPACITY_USED),
+    namespace: CEPH_STORAGE_NAMESPACE,
+  });
+  const poolRawCapacity: PoolMetrics = getPerPoolMetrics(
+    poolRawCapacityMetrics,
+    rawCapLoadError,
+    rawCapLoading,
+  );
```

Comment:
@aruniiird New metrics code for fetching metrics in a one request, you have to call same for your query